src/common/common-cmp-private.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2009-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "common-prelude.h"
+
+#ifndef MONGO_C_DRIVER_COMMON_CMP_PRIVATE_H
+#define MONGO_C_DRIVER_COMMON_CMP_PRIVATE_H
+
+
+#include <bson/bson.h> /* ssize_t, BSON_CONCAT */
+
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+
+BSON_BEGIN_DECLS
+
+
+/* Based on the "Safe Integral Comparisons" proposal merged in C++20:
+ * http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p0586r2.html
+ *
+ * Due to lack of type deduction in C, relational comparison functions (e.g.
+ * `cmp_less`) are defined in sets of four "functions" according to the
+ * signedness of each value argument, e.g.:
+ *  - mcommon_cmp_less_ss (signed-value, signed-value)
+ *  - mcommon_cmp_less_uu (unsigned-value, unsigned-value)
+ *  - mcommon_cmp_less_su (signed-value, unsigned-value)
+ *  - mcommon_cmp_less_us (unsigned-value, signed-value)
+ *
+ * Similarly, the `in_range` function is defined as a set of two "functions"
+ * according to the signedness of the value argument:
+ *  - mcommon_in_range_signed (Type, signed-value)
+ *  - mcommon_in_range_unsigned (Type, unsigned-value)
+ *
+ * The user must take care to use the correct signedness for the provided
+ * argument(s). Enabling compiler warnings for implicit sign conversions is
+ * recommended.
+ */
+
+
+#define MCOMMON_CMP_SET(op, ss, uu, su, us)                                              \
+   static BSON_INLINE bool BSON_CONCAT3 (mcommon_cmp_, op, _ss) (int64_t t, int64_t u)   \
```

Comment:
Common headers suggest `mcommon_*` entities should be defined in terms of `COMMON_NAME(*)` to support `MCOMMON_NAME_PREFIX`, e.g. using `mcommon_thread_create` as reference:
```c
#define MCOMMON_CMP_SET(op, ss, uu, su, us) \
   static BSON_INLINE bool COMMON_NAME (BSON_CONCAT3 (cmp_, op, _ss)) (...) { ... } \
   ...

#define mcommon_cmp_equal_ss COMMON_NAME (cmp_equal_ss)
```
Disregard if this is considered unnecessary for these headers.

Reply:
I expect the unique prefix is not needed for inline functions. I expect the prefix is used to prevent issues with duplicate symbols. With a missing prefix, Windows [builds failed](https://spruce.mongodb.com/task/mongo_c_driver_sasl_matrix_winssl_sasl_sspi_winssl_windows_2019_vs2017_x86_compile_patch_b6d64e58c4a1de2f554ac9e6dd20da223d80555e_6706e49b32bc5700071c2837_24_10_09_20_16_28/logs?execution=0) with:

```
error LNK2019: unresolved external symbol __mcd_emul_atomic_int64_fetch_add referenced in function _mcd_atomic_int64_fetch
```

Other inline functions in common headers (e.g. `bson_shared_mutex_init`) are similarly not prefixed.